limesurvey/tests/test_limesurvey.py

```diff
@@ -0,0 +1,476 @@
+
+        with self.assertRaises(ExceededLoginAttempts):
+            self.xblock.set_session_key()
+
+    def test_login_attempt_not_exceeded(self):
+        """
+        Check that when login attempt is exceeded an exception is raised.
+
+        Expected result:
+            - The exception is raised.
+        """
+        new_session_key = "test-session-key-1"
+        self.xblock.call_procedure = Mock(return_value=new_session_key)
+        self.xblock.get_survey_summary = Mock(side_effect=InvalidSessionKey)
+        self.xblock.last_login_attempt = datetime.now().replace(tzinfo=pytz.utc) - timedelta(days=1)
+
+        self.xblock.set_session_key()
+
+        self.xblock.call_procedure.assert_called_once_with(
+            "get_session_key",
+            settings.LIMESURVEY_API_USER,
+            settings.LIMESURVEY_API_PASSWORD,
+            get_session_key=True,
+        )
+        self.assertEqual(self.xblock.session_key, new_session_key)
+
+    def test_get_survey_summary(self):
+        """
+        Check getting the survey summary.
+
+        Expected result:
+            - A request is made to the LimeSurvey API getting the survey summary.
+        """
+        expected_result = "survey_summary"
+        self.xblock.call_procedure = Mock(return_value=expected_result)
+        params = {
+            "survey_id": self.xblock.survey_id,
+        }
+
+        result = self.xblock.get_survey_summary()
+
+        self.xblock.call_procedure.assert_called_once_with("get_summary", *params.values())
+        self.assertEqual(expected_result, result)
+
+    def test_get_student_access_code(self):
+        """
+        Check getting the current student access code given an anonymous_user_id.
+
+        Expected result:
+            - A request is made to the LimeSurvey API getting the access code.
+        """
+        expected_result = {"token": "test-token"}
+        self.xblock.call_procedure = Mock(return_value=expected_result)
+        anonymous_user_id = "test-anonymous-user-id"
+
+        self.xblock.set_student_access_code(anonymous_user_id)
+
+        self.xblock.call_procedure.assert_called_once_with(
+            "get_participant_properties",
+            self.xblock.survey_id,
+            {"attribute_1": anonymous_user_id},
+        )
+        self.assertEqual(expected_result.get("token"), self.xblock.access_code)
+
+    @data(
+            ("FirstName", "LastName", "FirstName LastName"),
+            ("LastName", "", "LastName"),
+            ("FirstName", "", "FirstName"),
+            ("", "", ""),
+    )
+    @unpack
+    def test_get_full_name(self, first_name, last_name, full_name):
+        """
+        Get student first name and last name given a user.
+
+        Expected result:
+            - Return a tuple with first and last name.
+        """
+        user = Mock(full_name=full_name, first_name=first_name, last_name=last_name)
```

Comment:
should the user instance have `first_name` and `last_name` attributes?

Reply:
It doesn't really matter. It's useful when checking if the method generated the values correctly (next lines)